### PR TITLE
chore: Update package tooling

### PR DESCRIPTION
**Pull request by jdpatt**
This updates the package tool around `pypi` just like in pyibisami.  Should be merged after #107 

- `pyproject.toml` is now the preferred way to define a [package's metadata, dependencies, and developer tool configuration](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/).  Updates setup.py to pyproject.toml
- Updates the `.gitignore` with the latest from [Github's gitignore templates](https://github.com/github/gitignore/blob/main/Python.gitignore)  
- Fixes already broken `tox` targets and updates them to the new src/ folder structure.
- Adds some developer tooling with `requirements-dev.txt` and `.pre-commit-config.yaml` which automate most the leg work of auto-formatting and etc..  You can read more about pre-commit on [their website](https://pre-commit.com/).
- Renames some of the files according to the discussion in #107.  

**Fixes**
- Icon loading in windows

### CODE REVIEW

**Comment by capn-freako** [comment]
**==> Merged.**

I used the `--squash` option in `merge` command, and am assuming that's why we have conflicts listed above.
(I hope I didn't just _really_ screw something up.)

Oh, actually, it looks like when I pulled David's branch `git` made some slight changes to his code:

```
capnf@DESKTOP-G84ND7C MINGW64 ~/Documents/GitHub/PyBERT (jdpatt-feature/update-package-tooling)
$ git log src/pybert/__main__.py
commit 07c9bb06b9caf03cdb00c5bca7b986a9ef8b535d (HEAD -> jdpatt-feature/update-package-tooling)
Author: David Banas <capn.freako@gmail.com>
Date:   Sun Oct 30 14:50:16 2022 -0700

    Pulled jdpatt:feature/update-package-tooling.

commit c8617fea8e173548cd5bb7012e4f11aed57b6721
Author: David Patterson <14203439+jdpatt@users.noreply.github.com>
Date:   Sat Oct 29 15:30:35 2022 -0400

    fix: update the imports to match the file renames

commit bf4570605fab24898614973ae2ff4441e31503fc
Author: David Patterson <14203439+jdpatt@users.noreply.github.com>
Date:   Mon Oct 17 21:32:10 2022 -0400

    refactor: move to a src python package

    This moves all source files from pybert to src/pybert. Some of the
    repeated pybert_pybert have been removed and reorganized.

$ git diff HEAD~ src/pybert/__main__.py
diff --git a/src/pybert/__main__.py b/src/pybert/__main__.py
index 6e56c43..1915358 100644
--- a/src/pybert/__main__.py
+++ b/src/pybert/__main__.py
@@ -1,10 +1,12 @@
 """Main entry into the PyBERT GUI."""
-from pybert.pybert      import PyBERT
 from pybert.gui.view import traits_view
+from pybert.pybert import PyBERT
+

 def main():
     thePyBERT = PyBERT()
     thePyBERT.configure_traits(view=traits_view)

-if __name__ == '__main__':
-       main()
+
+if __name__ == "__main__":
+    main()
```

Is that the _merge_ half of the `git pull` command trying to be helpful in resolving conflicts?

**Comment by capn-freako** [comment]
Hmmm, that's unfortunate, the _Closed with unmerged commits_ notice above is completely misleading.
(As is the _Closed_, as opposed to _Merged_ status of this PR. :( )
So, is this just the price we have to pay for squashing the incoming commits (Boy! I sure hope not.), or did I screw something up?

File: setup.py [hunk 1]
```diff
@@ -1,62 +0,0 @@
-""" setup.py - Distutils setup file for PyBERT package
```
Comment (capn-freako) [inline setup.py:1]: Did you mean to delete this file?
Is it that `setup.py` becomes superfluous once you introduce a `pyproject.toml`?
Is that true for _all_ users, regardless of how out-of-date their own local Python infrastructure may be?
That is, are we _certain_ that no one needs the `setup.py` file anymore?

**Review by capn-freako — COMMENTED** [review]
Missed this; caught it while reviewing #109 .

**Comment by jdpatt** [comment]
> Hmmm, that's unfortunate, the _Closed with unmerged commits_ notice below is completely misleading. (As is the _Closed_, as opposed to _Merged_ status of this PR. :( ) So, is this just the price we have to pay for squashing the incoming commits (Boy! I sure hope not.), or did I screw something up?

I do see a few things like the sim folder still exists which we renamed to `models` which also exists.  I'll keep digging but you can likely delete the sim folder.

File: setup.py (hunk 1 above)
Comment (jdpatt) [reply setup.py:1]: That is a very good question, you are currently supporting python 3.8 in conda.  I'm testing 3.9/3.10 for pypi/setuptools support.  As long as their version of pip is within the past few years, it should know what a `pyproject.toml` file is and tell the user to update pip to use it.  The file itself specs the version of setuptools that it requires to install it.

Overall, it is meant to be one or the other not both.  I think you actually get some errors and warnings if you have both.

Comment (capn-freako) [reply setup.py:1]: Okay, given the magnitude of this update, as well as the fact that we're updating the major version number (alerting people to potential breakage of existing client code and/or build infrastructure if I understand versioning protocol correctly), let's leave `setup.py` behind and just go for it, fully committing to `pyproject.toml`.

Cross your fingers we don't get severely bitten! ;-)

**Pull request merged**